src/app/project/board/contacts-tab-view/contacts-tab-view.component.ts

```diff
@@ -1,11 +1,48 @@
-import { Component, Input } from '@angular/core';
+import { Component, Input, OnInit } from '@angular/core';
+import { IId, Organization, Person } from '@dasch-swiss/dsp-js';
 
 @Component({
     selector: 'app-contacts-tab-view',
     templateUrl: './contacts-tab-view.component.html',
     styleUrls: ['./contacts-tab-view.component.scss']
 })
-export class ContactsTabViewComponent {
+export class ContactsTabViewComponent implements OnInit {
     // contact details
-    @Input() contactDetails: object;
+    @Input() contactDetails: Person[] | Organization[] | IId[];
+
+    @Input() subProperties: Object;
+
+    contactsList = [];
+    contactType: string;
+
+    ngOnInit() {
+
+        if (this.contactDetails) {
+            // check which type of array is present
+            this.contactType = this.getContactType(this.contactDetails[0]);
+
+            // if contactType is person or organization
+            if (this.contactType) {
+                this.contactsList = this.contactDetails;
+                return;
+            }
+
+            // if contactType is undefined, it means it contains the array of IId objects
+            for (const contact of this.contactDetails) {
+                this.contactsList.push(this.subProperties[contact.id]);
+            }
+            this.contactType = this.getContactType(this.contactsList[0]);
+
+        }
+    }
+
+    getContactType(obj: Person | Organization | IId) {
+        if (obj instanceof Person) {
+            return 'person';
+        }
+        if (obj instanceof Organization) {
+            return 'organization';
+        }
+        return undefined;
+    }
```

Comment:
method return type is missing and `if...else`

Since this is doing the same as `getAgentType()`, but the name is different, maybe it could be moved to some service under generic name to reuse in both cases?

Reply:
same for linter for else block